pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1639,123 @@ func TestValidateRouteUpdate(t *testing.T) {
+		{description: "missing parentheses or braces", input: `%{req.hdr(host)}`, validInput: false},
+		{description: "parameters for a sample fetch that doesn't take parameters", input: `%[ssl_c_der(host)]`, validInput: false},
+		{description: "dangerous sample fetches and converters", input: `%[env(FOO)]`, validInput: false},
+		{description: "dangerous sample fetches and converters", input: `%[req.hdr(host),debug()]`, validInput: false},
+		{description: "extra comma", input: `%[req.hdr(host),,lower]`, validInput: false},
+
+		// CR and LF are not allowed in header value as per RFC https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.4
+		{description: "linefeed", input: "\r", validInput: false},
+		{description: "CRLF", input: "\r\n", validInput: false},
+
+		// This value is allowed in haproxy.config and does not cause haproxy to crash. The input strings are sanitized by escaping single quotes in the value and then
+		// single quoting the whole value which helps haproxy not to crash in the event of an invalid string provided.
+		{description: "environment variable with a bracket missing", input: "${NET_COOLOCP_HOSTPRIMARY", validInput: true},
+		{description: "value with conditional expression and env var", input: `%[XYZ.hdr(Host)] if ${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
+		{description: "value with what looks like a conditional expression and env var", input: `%[XYZ.hdr(Host)]\ if\ ${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
+
+		// The syntax is http-request set-header <name> <fmt> [ { if | unless } <condition> ]. As per  As per this http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-request%20add-header
+		// the format  follows the log-format rules (see Custom Log Format in section 8.2.4) i.e http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.4
+		// The usage of it shall be as per section 8.2.6
+		// However for dynamic values, req.hdr, res.hdr, ssl_c_der are the allowed fetchers and lower, base64 are the allowed converters. So the following cases will fail.
+		{description: "sample value", input: "%ci:%cp [%tr] %ft %ac/%fc %[fc_err]/\\%[ssl_fc_err,hex]/%[ssl_c_err]/%[ssl_c_ca_err]/%[ssl_fc_is_resumed] \\%[ssl_fc_sni]/%sslv/%sslc", validInput: false},
+		{description: "interpolation of T i.e %T", input: `%T`, validInput: false},
+
+		// url
+		// regex does not check validity of url in a header value.
+		{description: "hdr with url", input: `%[XYZ.hdr(Host)] http:??//url/hack`, validInput: true},
+		{description: "hdr with url", input: `http:??//url/hack`, validInput: true},
+
+		// spaces and tab
+		// regex allows spaces before and after. The reason is that after a dynamic value is provided someone might provide a condition `%[XYZ.hdr(Host)] if foo` which would have
+		// spaces after the dynamic value and if condition.
+		// tab is rejected as control characters are not allowed by the regex.
+		{description: "space before and after the value", input: ` T `, validInput: true},
```

Comment:
Haven't we already tested for this using the examples that use ` if ` in the input value? What do we gain here?

Reply:
yes but wanted to add a test case just to add the unit test specifically to test the spaces.